### PR TITLE
Add EXECUTION_ENV_PATH localization for python backend

### PR DESCRIPTION
**Pull request by kthui**

### CODE REVIEW

File: src/filesystem.cc [hunk 1]
```diff
@@ -1293,6 +1325,15 @@ ASFileSystem::LocalizeDirectory(
   return DownloadFolder(container, object, dest);
 }
 
+Status
+ASFileSystem::LocalizeFile(
+    const std::string& path, std::shared_ptr<LocalizedDirectory>* localized)
+{
+  return Status(
```
Comment (Tabrizian) [inline src/filesystem.cc:1332]: Should we document in Python backend that only S3 is supported right now?

File: src/filesystem.cc [hunk 2]
```diff
@@ -1970,6 +2014,74 @@ S3FileSystem::LocalizeDirectory(
   return Status::Success;
 }
 
+Status
+S3FileSystem::LocalizeFile(
+    const std::string& path, std::shared_ptr<LocalizedDirectory>* localized)
+{
+  // Check if file exists
+  bool exists;
+  RETURN_IF_ERROR(FileExists(path, &exists));
```
[the remark this answers is not in the document]
Comment (kthui) [reply src/filesystem.cc:2023]: Yes. Bad default value for 'is_dir'. Updated.

File: src/backend_model.cc [hunk 3]
```diff
@@ -79,6 +79,11 @@ TritonModel::Create(
   std::shared_ptr<LocalizedDirectory> localized_model_dir;
   RETURN_IF_ERROR(LocalizeDirectory(model_path, &localized_model_dir));
 
+  // Localize paths in backend model config
+  // FIXME: Remove once a more permanent solution is implemented
```
[the remark this answers is not in the document]
Comment (kthui) [reply src/backend_model.cc:83]: Sure. Updated.

File: src/filesystem.cc (hunk 1 above)
Comment (kthui) [reply src/filesystem.cc:1332]: Yes. I can do the update. Maybe we can mark the feature as experiment for now, until a more permanent solution is implemented.

File: src/filesystem.cc [hunk 4]
```diff
@@ -144,6 +144,9 @@ class FileSystem {
   virtual Status LocalizeDirectory(
       const std::string& path,
       std::shared_ptr<LocalizedDirectory>* localized) = 0;
+  virtual Status LocalizeFile(
+      const std::string& path,
+      std::shared_ptr<LocalizedDirectory>* localized) = 0;
```
[the remark this answers is not in the document]
Comment (kthui) [reply src/filesystem.cc]: Sure. Updated.

**Comment by kthui** [comment]
Rebased. Change is from [aa0423a](https://github.com/triton-inference-server/core/pull/125/commits/aa0423a9d9cf80943e58e2f130a1ac7ffc56ca84) to [d1c3799](https://github.com/triton-inference-server/core/pull/125/commits/d1c37994df85647374c88335d6744c1390d74aa4)

**Review by GuanLuo — APPROVED** [review]

**Review by Tabrizian — APPROVED** [review]

**Pull request merged**